Tribler/community/trustchain/block.py

```diff
@@ -277,67 +248,69 @@ def sign(self, key):
         self.signature = crypto.create_signature(key, self.pack(signature=False))
 
     @classmethod
-    def create(cls, database, public_key, link=None):
+    def create(cls, transaction, database, public_key, link=None, link_pk=None):
         """
-        Create next block. Initializes a new block based on the latest information in the database
-        and optional linked block
+        Create an empty next block.
         :param database: the database to use as information source
+        :param transaction: the transaction to use in this block
         :param public_key: the public key to use for this block
         :param link: optionally create the block as a linked block to this block
+        :param link_pk: the public key of the counterparty in this transaction
         :return: A newly created block
         """
         blk = database.get_latest(public_key)
         ret = cls()
         if link:
-            ret.up = link.down
-            ret.down = link.up
             ret.link_public_key = link.public_key
             ret.link_sequence_number = link.sequence_number
         if blk:
-            ret.total_up = blk.total_up + ret.up
-            ret.total_down = blk.total_down + ret.down
             ret.sequence_number = blk.sequence_number + 1
             ret.previous_hash = blk.hash
-        else:
-            ret.total_up = ret.up
-            ret.total_down = ret.down
+        if link is None:
+            ret.link_public_key = link_pk
+
+        ret.transaction = transaction
         ret.public_key = public_key
         ret.signature = EMPTY_SIG
         return ret
 
-    def pack(self, data=None, offset=0, signature=True):
+    def pack(self, signature=True):
         """
         Encode this block for transport
-        :param data: optionally specify the buffer this block should be packed into
-        :param offset: optionally specifies the offset at which the packing should begin
         :param signature: False to pack EMPTY_SIG in the signature location, true to pack the signature field
         :return: the buffer the data was packed into
         """
-        buff = data if data else bytearray(block_pack_size)
-        pack_into(block_pack_format, buff, offset, self.up, self.down, self.total_up, self.total_down, self.public_key,
-                  self.sequence_number, self.link_public_key, self.link_sequence_number, self.previous_hash,
-                  self.signature if signature else EMPTY_SIG)
-        return str(buff)
+        encoded_tx = encode(self.transaction)
+
+        buff = bytearray(block_pack_size)
+        pack_into(block_pack_format, buff, 0, self.public_key, self.sequence_number, self.link_public_key,
+                  self.link_sequence_number, self.previous_hash, self.signature if signature else EMPTY_SIG)
+        return str(encoded_tx) + str(buff)
```

Comment:
For validation purposes please structure this as `str(buff) + struct.pack("I", str(encoded_tx)) + str(encoded_tx)` (or something like it).